platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+- Remove the `.babelrc` forms-library alias & custom SASS importer
+
+### Code Location
+
+The forms library package will live in `src/platform/packages/forms-library`.
+Since it will be a Node package, it will be added to the `package.json` file and apps will import it as a regular Node package instead of by its direct path.
+
+### Testing Plan
+
+We have ESLint rules in place which can verify that the right JS is imported as it relates to the forms library.
+Additionally, many of the end-to-end tests will fail if the forms library is not imported properly, so they will help with awareness.
+
+We don't have any automation in place to validate that the right SASS files are imported (other than webpack failing to resolve the file), so testing that forms styling is working properly will be a manual process using the browser dev tools.
+
+### Logging
+
+N/A
+
+### Debugging
+
+Since this project isn't about changing functionality of the forms library, we don't need to add any debugging tools. Validating that the app code can still access the forms library in its new location will be enough.
+
+### Caveats
+
+Though we will be creating a package for the forms library, we will not be following semantic versioning since it is _just_ a package and not a published module.
+The additional overhead of updating the forms-library's `package.json` file and updating the `yarn.lock` file in `vets-website` to target the newest (but unpublished) version seems like a burden without much benefit.
+
+### Security Concerns
+
+N/A
+
+### Privacy Concerns
+
+N/A
+
+### Open Questions and Risks
+
+Do we want to separate the forms library build & testing stages from the main `vets-website` build? By easing the load of what webpack and mocha are doing, we could speed things up. The downside is that since we wouldn't be running the forms library tests with every merge into master, we would have to have some other automation in place to make sure that any changes to `src/platform/packages/forms-library` passes the forms library tests.
+
+### Work Estimates
+
+#### Prep work
+
+This involves removing unused forms code and rearranging things so that the forms library doesn't import from app code.
+
+**Estimate:** < 1 day
+
+#### Creating the package
+
+This captures the bulk of the work. Consolidating the forms library code into a single location, fixing broken imports, updating babel and webpack configuration.
+
+**Estimate**: < 4 days
+
+#### Creating the boundaries
+
+Once the package is created, there will still be pieces of code that will need to be shifted around in order to firmly solidfy the boundaries around the forms library.
+
+**Estimate:** < 1 week
+
+#### Updating app code
+
+By this point, the package will already exist, but due to some babel and webpack config it will be masquerading as if it were still spread across `platform/forms` and `platform/forms-system`. This is where each app team updates their code to import directly from the new package.
```

Comment:
Should it be up to the app teams? Can we automate this with a simple find-and-replace? A codemod?